### PR TITLE
update riscv toolchain to 8.2

### PR DESCRIPTION
**Pull request by kaspar030**
The current risc-v toolchain makes some applications not pass the tests.
This gcc 8.1 based one works fine. I didn't dig for reasons. My gut feeling is that the newly available newlib-nano saves enough stack space to not break the tests.

Supercedes #62.
~~Waiting for #70.~~

### CODE REVIEW

**Comment by aabadie** [comment]
There's already a [new release](https://github.com/gnu-mcu-eclipse/riscv-none-gcc/releases/tag/v8.2.0-2.2-20190521) of the toolchain. Maybe use this one ?

**Comment by kaspar030** [comment]
- changed to 8.2
- parameterized toolchain version to easy future updates
- changed commit message to reflect new version

File: Dockerfile [hunk 1]
```diff
@@ -130,21 +130,21 @@ ENV MIPS_ELF_ROOT /opt/mips-mti-elf/2016.05-03
...
     echo 'Deduplicating binaries' >&2 && \
     cd /opt/gnu-mcu-eclipse/riscv-none-gcc/*/riscv-none-embed/bin && \
-    for f in *; do rm "$f" && ln "../../bin/riscv-none-embed-$f" "$f"; done && cd -
+      for f in *; do test -f "../../bin/riscv-none-embed-$f" && \
+       ln -f "../../bin/riscv-none-embed-$f" "$f"; \
+      done && \
+    cd -
 
-# HACK download arch linux' flex dynamic library
-RUN wget -q https://sgp.mirror.pkgbuild.com/core/os/x86_64/flex-2.6.4-2-x86_64.pkg.tar.xz -O- \
```
Comment (fjmolinas) [inline Dockerfile]: Why was this needed before and why is is not needed anymore?

Comment (kaspar030) [reply Dockerfile]: See #39, libfl on ubuntu is somehow weird. the riscv toolchain could not use it. thus we used arch linux' libfl package.
The current risc-v toolchain comes bundled with libfl, thus the hack is not needed anymore.

**Comment by kaspar030** [comment]
Thanks @fjmolinas! &go.
